website/status_screen/templates/status_screen/status_screen.html

```diff
@@ -0,0 +1,185 @@
+                    lastRefresh: null,
+                }
+            },
+            created() {
+                this.refresh();
+                this.recalculate_progress_interval = setInterval(this.updateTrackProgress, 100);
+                document.addEventListener("visibilitychange", this.visibilityChange);
+            },
+            unmounted() {
+                document.removeEventListener("visibilitychange", this.visibilityChange);
+            },
+            computed: {
+                orders_in_progress_grouped_user() {
+                    return this.map_orders_by_user(this.orders_in_progress);
+                },
+                orders_ready_grouped_user() {
+                    return this.map_orders_by_user(this.orders_ready);
+                },
+                orders_in_progress() {
+                    return this.orders.filter(order => {
+                        return order.ready === false;
+                    });
+                },
+                orders_ready() {
+                    return this.orders.filter(order => {
+                        return order.ready === true;
+                    });
+                },
+            },
+            methods: {
+                map_orders_by_user(orders) {
+                    let orders_mapped_user = {};
+                    for (let i = 0; i < orders.length; i++) {
+                        let order = orders[i];
+                        let user_id = order['user']['id'];
+                        let user = order['user'];
+                        if (user_id in orders_mapped_user) {
+                            orders_mapped_user[user_id]['orders'].push(order);
+                        } else {
+                            orders_mapped_user[user_id] = {
+                                'orders': [order],
+                                'user': user,
+                            }
+                        }
+                    }
+                    return orders_mapped_user;
+                },
+                visibilityChange(event) {
+                    if (event.target.hidden) {
+                        clearTimeout(this.refreshTimer);
+                    } else {
+                        clearTimeout(this.refreshTimer);
+                        if (this.lastRefresh === null || (new Date()).getTime() - this.lastRefresh > 5000) {
+                            this.refresh();
+                        } else {
+                            this.refreshTimer = setTimeout(this.refresh, 5000);
+                        }
+                    }
+                },
+                refresh() {
+                    if (this.refreshing) {
+                        return;
+                    }
+
+                    clearTimeout(this.refreshTimer);
+                    this.refreshing = true;
+                    return fetch(
+                        `/api/v1/shifts/{{ shift.id }}/orders/?type=0&picked_up=false&ordering=ready_at`,
```

Comment:
Is there maybe a way to use a {% url %} template tag here?

Like in https://github.com/JobDoesburg/yivi-portal/blob/51ced900cbda534e763f61403eb27eb5251740e6/schememanager/templates/yivi_request.html#L36

Reply:
And similarly on all other places, actually

Reply:
@KiOui status update?